### PR TITLE
Feature/add star rating in smartbanner

### PR DESCRIPTION
**Pull request by siscu**
Tenemos la necesidad desde motor que en el smartbanner pueda aparecer un rating, 
hacemos una primera aproximación, estando pendiente de validación.

image

### CODE REVIEW

File: components/ad/smartbanner/src/rating-star.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      typeStar = 'empty'
+    }
+    const SvgIconsCall = svgIconsType[typeStar]
+
+    return (<SvgIconsCall key={index} svgClass='sui-AdSmartbanner-ratingIcon' />)
+  }
+
+  return (
+    <span className='sui-AdSmartbanner-rating'>
+      {Array(ratingMax).fill().map(_renderStar)}
```
Comment (JoanClaret) [inline components/ad/smartbanner/src/rating-star.js:43]: awesome! great idea 🎉

File: components/ad/smartbanner/src/index.js [hunk 2]
```diff
@@ -18,22 +17,24 @@ class AdSmartbanner extends Component {
   }
 
   render () {
-    const { imageUrl, title, text, buttonText, staticPosition } = this.props
+    const { imageUrl, title, text, buttonText, staticPosition, icon, ratingValue, ratingMax, customRatingIcons } = this.props
+    const IconClose = icon || IconCloseDefault
```
Comment (jcger) [inline components/ad/smartbanner/src/index.js]: You could set it as defaultProp

**Review by JoanClaret — APPROVED** [review]
Looks great to me. Good job!

File: components/ad/smartbanner/src/index.js (hunk 2 above)
Comment (siscu) [reply components/ad/smartbanner/src/index.js]: Ok, added changes.

**Review by naxhh — APPROVED** [review]
NIce job!

But I wodering why we didn't went with composable elementes...

Like the rating being a component and allowing to be injected into the banner

That way, for example, the rating* props are not coupled into the banner

And people not using the rating system has not that dependency on their bundle also.

**Comment by siscu** [comment]
@naxhh I was talking with Kiko and he suggested that would be much better to make a first interaction before have a UX definition. If that works ok for you?

**Review by kikoruiz — APPROVED** [review]
Que bien enseñado que vienes ya!! 👏

File: components/ad/smartbanner/src/rating-star.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import IconStarFullDefault from '@schibstedspain/sui-svgiconset/lib/Star'
+import IconStarHalfDefault from '@schibstedspain/sui-svgiconset/lib/Starhalf'
+import IconStarEmptyDefault from '@schibstedspain/sui-svgiconset/lib/Starempty'
+
+const baseNumber = 0.5
+const checkRatingValue = ({ ratingValue, ratingMax }) =>
+  (ratingValue % baseNumber === 0) && ratingValue <= ratingMax
+
+const RatingStar = ({ratingValue, ratingMax, icons}) => {
+  const IconStarFull = icons.starFull || IconStarFullDefault
```
Comment (jcger) [inline components/ad/smartbanner/src/rating-star.js:11]: this could be defaults too

**Review by jcger — APPROVED** [review]

File: components/ad/smartbanner/src/index.js [hunk 4]
```diff
@@ -44,16 +44,21 @@ class AdSmartbanner extends Component {
...
   onClose: PropTypes.func.isRequired,
-  icon: PropTypes.func.isRequired,
+  icon: PropTypes.func,
   imageUrl: PropTypes.string.isRequired,
   title: PropTypes.string.isRequired,
   text: PropTypes.string.isRequired,
   buttonText: PropTypes.string.isRequired,
-  staticPosition: PropTypes.bool
+  staticPosition: PropTypes.bool,
+  customRatingIcons: PropTypes.obj,
```
Comment (JoanClaret) [inline components/ad/smartbanner/src/index.js:53]: What about specifying the object shape here?

File: components/ad/smartbanner/src/rating-star.js (hunk 3 above)
Comment (siscu) [reply components/ad/smartbanner/src/rating-star.js:11]: Thanks Julian. Fixed.

File: components/ad/smartbanner/src/index.js (hunk 4 above)
Comment (siscu) [reply components/ad/smartbanner/src/index.js:53]: Ok, added.

Comment (kikoruiz) [reply components/ad/smartbanner/src/index.js:53]: Pero el shape ya está definido en el componente que requiere la prop, aquí en realidad no es necesario.

**Pull request merged**